File: RB101-109/Easy3/ex5.rb
# Using the multiply method from the "Multiplying Two Numbers" problem, write a 
# method that computes the square of its argument (the square is the result of 
# multiplying a number by itself).

# Example:
# square(5) == 25
# square(-8) == 64

def square(int)
  multiply(int,int)
end

def power_to_the_n(int, exponent)
  total = 1
  exponent.times{ |n| total = multiply(total,int) }
  return total
end
  
def multiply(int1, int2)
  int1 * int2
end

puts square(5) == 25
puts square(-8) == 64
puts power_to_the_n(2,4)
puts power_to_the_n(3,3)
puts power_to_the_n(1, 100)
puts power_to_the_n(10, 0)
puts power_to_the_n(0, 10)

File: JS210/Basics/ex5.js
// Write a program that prompts the user for two positive integers, and then
// prints the results of the following operations on those two numbers: addition,
// subtraction, product, quotient, remainder, and power.
// Do not worry about validating the input.

// Example
// ==> Enter the first number:
// 23
// ==> Enter the second number:
// 17
// ==> 23 + 17 = 40
// ==> 23 - 17 = 6
// ==> 23 * 17 = 391
// ==> 23 / 17 = 1
// ==> 23 % 17 = 6
// ==> 23 ** 17 = 1.4105003956066297e+23

// The final output above shows how JavaScript displays numbers that are too
// large for its Number type. The number represents approximately
// 1.41 * 100,000,000,000,000,000,000,000 using what is called floating-point
// notation. If you want to see what the exact value is, you can use
// JavaScript's BigInt type by appending an n to both 23 and 17:

// > 23n ** 17n    // 141050039560662968926103n

let rlSync = require('readline-sync');
let int1 = Number(rlSync.question('What is the first number?\n'));
let int2 = Number(rlSync.question('What is the second number?\n'));
console.log(`${int1} plus ${int2} = ${int1 + int2}`);
console.log(`${int1} minus ${int2} = ${int1 - int2}`);
console.log(`${int1} times ${int2} = ${int1 * int2}`);
console.log(`${int1} divided by ${int2} = ${int1 / int2}`);
console.log(`${int1} to the ${int2} power = ${int1 ** int2}`);
console.log(`${int1} remainder ${int2} = ${int1 % int2}`);
